runtime/libia2/include/permissive_mode.h

```diff
@@ -470,7 +470,13 @@ __attribute__((constructor)) void permissive_mode_init(void) {
 // so we need to free what `getline` allocated with `__real_free`.
 void __real_free(void *ptr);
 
-extern uintptr_t ia2_stack_addrs[MAX_PKEYS];
+typedef struct {
+  pid_t tid;
+  uintptr_t stack_addrs[MAX_PKEYS];
+} IA2ThreadData;
```

Comment:
We should not define this struct twice so they don't get out of sync. I'm aware this is also a potential issue with the types in the variable declarations you used for labeling the other parts, but if we're defining an entire struct it's more likely we'll end up changing it so it's a bigger issue.

Reply:
Yeah, I'd definitely rather do that.  Is there a good place to put this?  I'd like to move the other `extern`s there, too.

Reply:
Moved to `ia2_threads.h` and `threads.c`.  The `ia2_*` is because `threads.h` clashes.